bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/quickaccess/providers/CommandProvider.java

```diff
@@ -82,6 +97,25 @@ public QuickAccessElement[] getElements() {
 			if (commandService == null) {
 				return null;
 			}
+
+			// initialize all the workbench commands to fqn.
+			if (idToFqn.isEmpty()) {
+				IExtensionRegistry registry = Platform.getExtensionRegistry();
+				IExtensionPoint point = registry.getExtensionPoint(IWorkbenchRegistryConstants.EXTENSION_COMMANDS);
+				IExtension[] extensions = point.getExtensions();
+				for (IExtension extension : extensions) {
+					IConfigurationElement[] elements = extension.getConfigurationElements();
+					for (IConfigurationElement element : elements) {
+						if (IWorkbenchRegistryConstants.TAG_COMMAND.equals(element.getName())) {
+							String id = element.getAttribute(IWorkbenchRegistryConstants.ATT_ID);
+							// plugin-id/command-id.
+							String qualifiedName = element.getContributor().getName() + "/" + id; //$NON-NLS-1$
+							idToFqn.put(id, qualifiedName);
+						}
+					}
+				}
+			}
```

Comment:
Extract this to a private method to improve readability (don't bloat the `getElements()` method)

Reply:
Fixed.